src/components/select/SelectItemCommons.tsx

```diff
@@ -42,13 +42,19 @@ function withSelectItemWrapper<P extends SelectOption>(
   optionType: SelectOptionType,
 ) {
   const Wrapper = forwardRef<HTMLDivElement, P>((props, ref) => {
-    const { helpText, label, prefix, suffix, value, ...rest } = props;
+    const { disabled, group, helpText, label, prefix, selected, suffix, value, ...rest } = props;
```

Comment:
here we want to pass all the options props to the WrappedComponent and options being a generic, I think we need to extract all Mantine props instead of all options props. Mantine props should be fixed on the contrary of options props.

Reply:
If I understand properly, I agree!
In other words, we want all the custom props that the consumer passes in the options to be forwarded to the wrapped component, but we don't want the technical props from Mantine to be.

HelpText, Prefix & suffix are already being used, so there's no reason to forward them either.

Reply:
Right, I'm effectively removing custom props here!
Is there a reference of the Mantine props passed to the `itemComponent` somewhere? [Here](https://v6.mantine.dev/core/select/?t=props) it says `any`..

Reply:
I don't think we have a reference, we need to check by ourselves what's passed down.